File: graph/model/post.go
package model

type Post struct {
	Title       string `json:"title"`
	Content     string `json:"content"`
	UUID        string `json:"uuid"`
	Created     int64  `json:"created"`
	Updated     int64  `json:"updated"`
	UpdateCount int64  `json:"updateCount"`
}

func (Post) IsKeywordLink() {}

type NewPost struct {
	Title      string   `json:"title"`
	Content    string   `json:"content"`
	WriterUUID string   `json:"writerUuid"`
	Keywords   []string `json:"keywords"`
}

type UpdatePost struct {
	UUID    string  `json:"uuid"`
	Title   *string `json:"title"`
	Content *string `json:"content"`
}

File: src/model/post.rs
use super::keyword::Keyword;

#[derive(GraphQLObject)]
pub struct Post {
    pub title: String,
    pub content: String,
    pub uuid: String,
    pub keywords: Vec<Keyword>,
    pub created: i32,
    pub updated: i32,
    pub update_count: i32
}

#[derive(GraphQLInputObject)]
pub struct NewPost {
    pub title: String,
    pub content: String,
}

#[derive(GraphQLInputObject)]
pub struct UpdatePost {
    pub uuid: String,
    pub name: String,
}
